### PR TITLE
Family fee voucher

### PR DESCRIPTION
**Pull request by Mudassar045**

### CODE REVIEW

File: web/src/modules/Student/Single/Fees/printPreview.tsx [hunk 1]
```diff
@@ -57,45 +50,90 @@ class printPreview extends Component <propTypes, S>{
...
 		}
 
-		return student.payments
+		return this.student().payments
 	}
 
-	render() {
+	student = (): MISStudent => {
+		const id = this.studentID()
+		return this.props.students[id] || this.siblings()[0]
```
Comment (taimur38) [inline web/src/modules/Student/Single/Fees/printPreview.tsx]: I think this should be `return id === undefined ? this.siblings()[0] : this.props.students[id]`

in the event that there is somehow an undefined key in the students object

File: web/src/modules/Student/Single/Fees/printPreview.tsx [hunk 2]
```diff
@@ -57,45 +50,90 @@ class printPreview extends Component <propTypes, S>{
...
+	}
 
-		const { classes, student, settings } = this.props
+	siblings = (): MISStudent[] => {
+		const famId = this.familyID()
+		return Object.values(this.props.students)
+			.filter(s => s && s.Name && s.FamilyID && s.FamilyID === famId)
+	}
```
Comment (taimur38) [inline web/src/modules/Student/Single/Fees/printPreview.tsx]: can we use memoization here? seems fairly expensive and we call it a decent amount of times.

Comment (taimur38) [reply web/src/modules/Student/Single/Fees/printPreview.tsx]: for now it's fine. but will assign you a task for next sprint to look into memoization

File: web/src/modules/Student/Single/Fees/StudentLedgerPage.tsx [hunk 3]
```diff
@@ -74,16 +95,11 @@ export const StudentLedgerPage : React.SFC < StudentLedgerPageProp > = ({ paymen
 				<div className={owed <= 0 ? "advance-amount" : "pending-amount"}><b>Rs. {numberWithCommas(Math.abs(owed))}</b></div>
 			</div>
 			{ // don't show if student ledger rendered in historical fee module, Voucher No. hard coded in this case
-				voucherNo !== 777 && <>
-					<div className="row voucher-signature line">
-						<div>___________________</div>
-						<div>___________________</div>
-					</div>	
+				voucherNo !== 777 &&
```
[the remark this answers is not in the document]
Comment (Mudassar045) [reply web/src/modules/Student/Single/Fees/StudentLedgerPage.tsx]: I used it as default, but will make it optional and then apply check accordingly

File: web/src/modules/Student/Single/Fees/printPreview.tsx (hunk 1 above)
Comment (Mudassar045) [reply web/src/modules/Student/Single/Fees/printPreview.tsx]: I got you.

File: web/src/modules/Student/Single/Fees/printPreview.tsx (hunk 2 above)
Comment (Mudassar045) [reply web/src/modules/Student/Single/Fees/printPreview.tsx]: Yes, we should use memoization here because state changing in the component

**Pull request merged**